### ISSUE TITLE
UCX Assesment Workflow breaks when executing Crawl permissions if Model Serving is not available in the Workspace Region

### ISSUE THREAD

**Issue by jarnawer** [databrickslabs/ucx#1228]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a series of Workspaces deployed in Switzerland North Azure Region. Due to regulatory compliance requirements it has to be that exact region.

When executing Assessment Workflow, it breaks in the "Crawl Permissions" step. The reason of the failure is because it is trying to access Model serving endpoints features to crawl permissions, but according to Databricks documentation, it is not enabled in Switzerland (https://learn.microsoft.com/en-us/azure/databricks/machine-learning/model-serving/model-serving-limits#--region-availability)

### Expected Behavior

Crawl permissions step, should execute regardless if Model Serving is available on the region or not. It should be configurable to crawl permissions for that.

### Cloud

Azure

### Operating System

Linux

### Version

latest via Databricks CLI

### Relevant log output

```shell
UCX v0.17.0
Scans the workspace-local groups and all their permissions. The list is stored in the `$inventory.permissions`
Delta table.

This is the first step for the _group migration_ process, which is continued in the `migrate-groups` workflow.
This step includes preparing Legacy Table ACLs for local group migration.


15:19:46  INFO [d.labs.ucx] UCX v0.17.0 After job finishes, see debug logs at /Workspace/Applications/ucx/logs/assessment/run-413746228294294/crawl_permissions.log
15:19:46  INFO [d.l.u.workspace_access.manager] Cleaning up inventory table hive_metastore.ucx.permissions
15:19:46  INFO [d.l.u.workspace_access.manager] Inventory table cleanup complete
15:19:46  INFO [d.l.u.workspace_access.generic] Listed clusters in 0:00:00.047167
15:19:46  INFO [d.l.u.workspace_access.generic] Listed cluster-policies in 0:00:00.038094
15:19:46  INFO [d.l.u.workspace_access.generic] Listed instance-pools in 0:00:00.049407
15:19:46  INFO [d.l.u.workspace_access.generic] Listed sql/warehouses in 0:00:00.037511
15:19:46  INFO [d.l.u.workspace_access.generic] Listed jobs in 0:00:00.162886
15:19:46  INFO [d.l.u.workspace_access.generic] Listed pipelines in 0:00:00.027808
15:19:46 ERROR [d.labs.ucx] Execute `databricks workspace export //Applications/ucx/logs/assessment/run-413746228294294/crawl_permissions.log` locally to troubleshoot with more details. Model serving is not enabled for your shard. Please contact your organization admin or Databricks support.
```

**Comment by nfx** [comment]
@jarnawer Please include the exact stack trace.

**Comment by jarnawer** [comment]
Hi @nfx, what exact stack trace do you mean?. I can provide the standard error output, but has the same information.

**Comment by nfx** [comment]
@jarnawer okay, the fix is simple: surround these lines with `try: ... except NotFound: pass` or something like that.

https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/workspace_access/generic.py#L58-L59

**Comment by RobMandersBJSS** [comment]
I'm sorry to say it does not appear this issue has been resolved.

I've attempted to re-run the assessment pipeline today following the release of v0.22.0 which includes the fix in PR #1275, however, the `crawl_permissions` jobs is still failing on the same 'FEATURE_DISABLED' error I was experiencing last week (v0.21.0).

This was a fresh installation of UCX and I have confirmed I am running the latest version:

![image](https://github.com/databrickslabs/ucx/assets/145695371/5d498de2-3738-4c83-8b14-974b654908af)
![image](https://github.com/databrickslabs/ucx/assets/145695371/1f4c7da4-9c6b-4f24-80f7-207669688654)

```json
// version.json
{
  "version": "0.22.0",
  "wheel": "/Applications/ucx/wheels/databricks_labs_ucx-0.22.0-py3-none-any.whl",
  "date": "2024-04-29T08:43:20.038438+00:00"
}
```

I've attached the full logs in `crawl_permissions.log`, here:

[crawl_permissions.log](https://github.com/databrickslabs/ucx/files/15147596/crawl_permissions.log)

Please let me know if there is anything you need or if I've missed some configuration.

Thank you.

**Comment by rafa-arana** [comment]
Just hit this in UCX v0.27.1

```
InternalError: Listing serving-endpoints failed: Model serving is not enabled for your shard. Please contact your organization admin or Databricks support.
---------------------------------------------------------------------------
InternalError                             Traceback (most recent call last)
File ~/.ipykernel/4656/command--1-3817622542:18
     15 entry = [ep for ep in metadata.distribution("databricks_labs_ucx").entry_points if ep.name == "runtime"]
     16 if entry:
     17   # Load and execute the entrypoint, assumes no parameters
---> 18   entry[0].load()()
     19 else:
     20   import importlib

File /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.11/site-packages/databricks/labs/ucx/runtime.py:103, in main(*argv)
    101 if len(argv) == 0:
    102     argv = sys.argv
--> 103 Workflows.all().trigger(*argv)

File /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.11/site-packages/databricks/labs/ucx/runtime.py:80, in Workflows.trigger(self, *argv)
     78 workflow = self._workflows[workflow_name]
     79 if task_name == "parse_logs":
---> 80     return ctx.task_run_warning_recorder.snapshot()
     81 # `{{parent_run_id}}` is the run of entire workflow, whereas `{{run_id}}` is the run of a task
     82 workflow_run_id = named_parameters.get("parent_run_id", "unknown_run_id")

File /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.11/site-packages/databricks/labs/ucx/installer/logs.py:203, in TaskRunWarningRecorder.snapshot(self)
    201     error_messages.append(message)
    202 if len(error_messages) > 0:
--> 203     raise InternalError("\n".join(error_messages))
    204 return log_records

InternalError: Listing serving-endpoints failed: Model serving is not enabled for your shard. Please contact your organization admin or Databricks support.
UCX v0.27.1

UCX v0.27.1
```
